src/core/pagerankGraph.test.js

```diff
@@ -168,6 +168,64 @@ describe("core/pagerankGraph", () => {
     });
   });
 
+  describe("totalOutWeight", () => {
+    it("errors on a modified graph", () => {
+      const eg = examplePagerankGraph();
+      eg.graph().addNode(NodeAddress.fromParts(["bad", "node"]));
+      expect(() =>
+        eg.totalOutWeight(NodeAddress.fromParts(["bad", "node"]))
+      ).toThrowError("has been modified");
+    });
+    it("errors on nonexistent node", () => {
+      const eg = examplePagerankGraph();
+      expect(() =>
+        eg.totalOutWeight(NodeAddress.fromParts(["nonexistent"]))
+      ).toThrowError("non-existent node");
+    });
+    function verifyOutWeights(pg: PagerankGraph) {
+      const outWeight: Map<NodeAddressT, number> = new Map();
+      for (const node of pg.graph().nodes()) {
+        outWeight.set(node, pg.syntheticLoopWeight());
+      }
+      function addWeight(node: NodeAddressT, weight: number) {
+        const newWeight = NullUtil.get(outWeight.get(node)) + weight;
+        outWeight.set(node, newWeight);
+      }
+      for (const {edge, weight} of pg.edges()) {
+        addWeight(edge.src, weight.toWeight);
+        addWeight(edge.dst, weight.froWeight);
+      }
+      for (const node of pg.graph().nodes()) {
+        expect(pg.totalOutWeight(node)).toEqual(outWeight.get(node));
+      }
+    }
+    it("computes outWeight correctly on the example graph", () => {
+      const eg = examplePagerankGraph();
```

Comment:
Thought: 
The edge evaluator (`defaultEvaluator`) for `examplePagerankGraph()` sets `froWeight` to 0, so if you removed [this line], would this test not still pass? Because the example graph's `dst` node has two out edges that are both set to 0, so it's `totalOutWeight` will be it's synthetic loop weight. So maybe use a different edge evaluater for this test if those assumptions are correct.  

[this line]: https://github.com/sourcecred/sourcecred/blob/eea041a0fe2ef3938273387a5a3394b98bc33adb/src/core/pagerankGraph.js#L175

Reply:
verified those assumptions above are correct => removing that line that accumulates `froWeights` to `edge.dst` has no effect on the outcome of the test. 

And the following seems to fix that => 

```javascript 
function examplePagerankGraph(
    edgeEvaluator = defaultEvaluator
  ): PagerankGraph {
    const g = advancedGraph().graph1();
    return new PagerankGraph(g, edgeEvaluator);
  }

it("computes outWeight correctly on the example graph", () => {
      const edgeEvaluator = (edge) => ({toWeight: 1, froWeight: 2});
      const eg = examplePagerankGraph(edgeEvaluator);
      verifyOutWeights(eg);
  });
```

Reply:
Great catch, thanks! Fixed.